src/Symfony/Component/Dotenv/Command/DebugCommand.php

```diff
@@ -78,25 +97,30 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 : sprintf('<fg=red>⨯</> %s', $envFile);
         }, $envFiles));
 
+        $nameFilter = $input->getArgument('name');
+
         $io->section('Variables');
         $io->table(
             array_merge(['Variable', 'Value'], $availableFiles),
-            $this->getVariables($availableFiles)
+            $this->getVariables($availableFiles, $nameFilter)
         );
 
         $io->comment('Note real values might be different between web and CLI.');
 
         return 0;
     }
 
-    private function getVariables(array $envFiles): array
+    private function getVariables(array $envFiles, ?string $nameFilter): array
     {
-        $vars = explode(',', $_SERVER['SYMFONY_DOTENV_VARS'] ?? '');
-        sort($vars);
+        $vars = $this->getAvailableVars();
 
         $output = [];
         $fileValues = [];
         foreach ($vars as $var) {
+            if (null !== $nameFilter && !(0 === strcasecmp($nameFilter, $var) || 0 === stripos($var, $nameFilter))) {
```

Comment:
Why `strcasecmp` and `stripos` ? The later should be enough.

```suggestion
            if (null !== $nameFilter && 0 !== stripos($var, $nameFilter)) {
```

Reply:
you are right @GromNaN , updated the code